### DISCUSSION TITLE
Is there a planned UI update for ghidra

### DISCUSSION THREAD

**Discussion by neilhuang007** [NationalSecurityAgency/ghidra#7334] [General]
For some reason, i think the major reverse engineering toolkits have very bad visual design that resembles the 1940s, is there some sort of UI update planned for ghidra?
Or was it just me unable to appreciate art and design
what do u guys think bout the ui

**Comment by dragonmacher** [comment]
Computer design was at its prime in the 40s.    I'd say we are more 90s style.    An aesthetic redesign at this point is likely way to expensive to actually happen.

**Comment by msm-code** [comment]
>Or was it just me unable to appreciate art and design what do u guys think bout the ui

It's an acquired taste. If you spend a few hours changing the default layout, adding missing keybindings, installing quality of life plugins and extensions, it actually starts being almost ergononic ;).

Honestly though, with dark theme, and aforementioned changes I think Ghidra design is not that bad. It could really use an UX specialist to polish some rough edges, and a few person-months spent on QoL improvements (why is the Python interpreter so ugly? Why can't I zoom listing in/out with ctrl+wheel? Why are the default keybindings so inconsistent, and so many useufl keybindings are missing? Why in most panels it's impossible to navigate by keyboard only? Why is that damn smart selection impossible to disable? That kind of things).

But I personally care about ergonomy, not about eye candy. If you want slick graphics like ImHex then yeah probably we're never getting that.

**Comment by neilhuang007** [comment]
Unfortunately, I was already planning to do that.
However, I recognize that user experience is crucial for this project. Right now, popular tools like IDA and Binary Ninja have notoriously poor (sometimes downright terrible) interfaces, so adopting a better UI could give us a major advantage. Improved design would also enhance clarity and reduce the time people spend digging through documentation just to find a tiny or hidden feature. Plus, it would make reverse engineering more accessible to newcomers in the field.

I have about five years of UI/UX design experience, and I’m hoping anyone interested in this idea might want to team up with me. If you see this discussion and would like to collaborate, let me know!

**Reply by neilhuang007** [reply]
We can probably jsut take the icons from jetbrains, and then use javaFX with html to revamp the feild, this is probably gonna take a few months for me along but should be quick if someone help

**Reply by LukeSerne** [reply]
A UI/UX update would certainly be helpful. However, I personally don't have any UX experience, so I think it's better left to someone else. Also, if a UI/UX update is started by the community, I think it will be very important to coordinate this with the Ghidra team to make it as easy as possible for them to review and merge all the changes that come from this project. It would be a shame if people put in the work to make a UI/UX update and the resulting PR(s) is/are never merged. I'm not sure what the best way to coordinate this is though.

**Reply by Wall-AF** [reply]
If you do update the UI/UX, PLEASE don't forget keyboard only users.  (The likes of Microsoft etc. have!!!)

**Comment by neilhuang007** [comment]
Guys im planning to contribute by updating the keybindings, what functionalities do we want keybinds for
